api/prisma/schema.prisma

```diff
@@ -7,40 +7,204 @@ datasource db {
+  challengeType Int? // Null | Undefined
   completedDate Float
   id            String
+  solution      String? // Null | Undefined
+
+  // Nonesense
+  dataSource Json?   @map("__dataSource") // Null
+  persisted  Boolean @map("__persisted")
+  strict     Boolean @map("__strict")
 }
 
-type UserProfileUi {
+type CompletedChallenge {
+  challengeType      String? // Null | Undefined
+  completedDate      Float
+  files              File[]
+  githubLink         String? // Undefined
+  id                 String
+  isManuallyApproved Boolean? // Undefined
+  solution           String? // Undefined
+
+  // Nonsense
+  cachedRelations Json?                   @map("__cachedRelations")
+  data            CompletedChallengeData?
+  dataSource      Json?                   @map("__dataSource") // Null only
+  persisted       Boolean?                @map("__persisted") // Undefined
+  strict          Boolean?                @map("__strict") // Undefined
+}
+
+type Portfolio {
+  description String
+  id          String
+  image       String
+  title       String
+  url         String
+}
+
+type ProfileUIData {
   isLocked      Boolean
   showAbout     Boolean
   showCerts     Boolean
-  showDonation  Boolean
+  showDonation  Boolean? // Undefined
   showHeatMap   Boolean
   showLocation  Boolean
   showName      Boolean
   showPoints    Boolean
   showPortfolio Boolean
   showTimeLine  Boolean
+
+  // Nonesense
+  persisted Boolean? @map("__persisted") // Undefined
+  strict    Boolean? @map("__strict") // Undefined
+}
+
+type ProfileUI {
+  isLocked      Boolean? // Undefined
+  showAbout     Boolean? // Undefined
+  showCerts     Boolean? // Undefined
+  showDonation  Boolean? // Undefined
+  showHeatMap   Boolean? // Undefined
+  showLocation  Boolean? // Undefined
+  showName      Boolean? // Undefined
+  showPoints    Boolean? // Undefined
+  showPortfolio Boolean? // Undefined
+  showTimeLine  Boolean? // Undefined
+
+  // Nonesense
+  cachedRelations Json?          @map("__cachedRelations") // Undefined | {}
+  data            ProfileUIData? @map("__data") // Undefined
+  persisted       Boolean?       @map("__persisted") // Undefined
+  strict          Boolean?       @map("__strict") // Undefined
+}
+
+type ProgressTimestamp {
+  timestamp Float
+}
+
+type SavedChallengeFile {
+  contents String
+  ext      String
+  history  String[]
+  key      String
+  name     String
+}
+
+type SavedChallenge {
+  files         SavedChallengeFile[]
+  id            String
+  lastSavedDate Float
 }
 
+model user {
+  id                           String               @id @default(auto()) @map("_id") @db.ObjectId
+  about                        String
+  acceptedPrivacyTerms         Boolean
+  badges                       Json? // Undefined | { coreTeam [][] }
+  completedChallenges          CompletedChallenge[]
+  currentChallengeId           String?
+  donationEmails               String[]             @default([]) // Undefined | String[]
```

Comment:
```suggestion
  donationEmails               String[]             @default([]) // Undefined | String[] (only possible for built in Types like String)
```
Because I keep forgetting.